### PR TITLE
Fixes for Student Mode (200, 207, 212)

### LINKED ISSUE #200: Description selector bug: All descriptions appear in dropdown instead of just quantities or equations
Equation selector should only have the equations as options, quantities should only have quantities as options.

### LINKED ISSUE #207: Quantities header change
The column header for quantities should count the number of unknowns and not include parameters.  The title should be “Unknowns: 1”

### LINKED ISSUE #212: Wording change re: Root
Current, there is a popup warning that says "No variable is marked as Root"  It should say "No quantity is marked as Sought."

Also, in the quantity node editor, please relabel the check box from Root to Sought.

### PR DESCRIPTION
**Pull request by dharanisakary**
Resolves #200 
Resolves #207 
Resolves #212

### CODE REVIEW

File: www/js/con-student.js [hunk 1]
```diff
@@ -304,12 +294,48 @@ define([
...
+				//return obj1.label > obj2.label;
+				if(descNameMap[obj1.value] && descNameMap[obj2.value])
+					return descNameMap[obj1.value].toLowerCase().localeCompare(descNameMap[obj2.value].toLowerCase());
+			}, this);
+			
+			d.addOption({label: "--Select--", value: "defaultSelect"});
+			array.forEach(descriptions, function (desc) {
+				if(desc.label !== undefined){
+					d.addOption(desc);
+					console.log("Description is : ", desc);
```
Comment (jonwetzel) [inline www/js/con-student.js]: Please remove development/test console logs when you're finished with them.

File: www/js/con-student.js [hunk 2]
```diff
@@ -304,12 +294,48 @@ define([
...
+					var name = this._model.authored.getName(desc.value);
+					var option = {label: name + " (" + desc.label + ")", value: desc.value};
+					console.log("option is",option);
```
Comment (jonwetzel) [inline www/js/con-student.js]: Another console log.

File: www/js/con-student.js [hunk 3]
```diff
@@ -304,12 +294,48 @@ define([
 			// Apply settings from PM
 			console.log("Initial Control Settings for Student");
 			
+			var d = registry.byId(this.controlMap.description);
+			
+			console.log("description widget = ", d, this.controlMap.description);
```
Comment (jonwetzel) [inline www/js/con-student.js]: Another console log.

File: www/js/con-student.js [hunk 4]
```diff
@@ -216,7 +213,7 @@ define([
 			//restore to default  - creating select input as stateless
 			registry.byId(this.controlMap.inputs).set('value', 'defaultSelect', false);
 			// Logging
-		},
+		},
```
Comment (jonwetzel) [inline www/js/con-student.js]: You can remove the unintentional extra spacing added here.

**Comment by jonwetzel** [comment]
Looks good, just need @sachingrover211 's review.

File: www/js/draw-model.js [hunk 5]
```diff
@@ -530,14 +539,12 @@ define([
...
 			}
 			domConstruct.destroy(nodeID);
-			if(this._model.getColor(ID) !== undefined){
-				if(type === 'quantity'){
-					this._quantityNodeCount--;
-				}else if( type === 'equation'){
-					this._equationNodeCount--;
-				}
-				this.addNodeCount();
+			if(type === 'quantity' && this._model.getVariableType(ID) === "unknown"){
```
Comment (sachingrover211) [inline www/js/draw-model.js]: you should be checking for dynamic as well in this. That field will be coming back in the code.

File: www/js/draw-model.js [hunk 6]
```diff
@@ -130,7 +133,12 @@ define([
 				else
 					this._borderColor = this._colors.indexOf(color) - 1;
 			}
-
+			if(node.type === 'quantity'){
+				if(this._model.getVariableType(node.ID) === "unknown")
```
Comment (sachingrover211) [inline www/js/draw-model.js]: @jonwetzel shouldn't these checks include unknown and dynamic?

File: www/js/draw-model.js [hunk 7]
```diff
@@ -189,6 +197,12 @@ define([
 					this.addNodeDescription(node.ID);
 				}
 			}
+			if(node.variableType === "unknown" && cachedNode.variableType !== "unknown"){
+				this._unknownQuantityNodeCount++;
```
Comment (sachingrover211) [inline www/js/draw-model.js]: Sorry this code shouldn't be here. The node counts should be a property of the model, so all counting related code should be there. The logic for whatever the count is. And then it should be shown in draw-model.

File: www/js/con-student.js [hunk 8]
```diff
@@ -304,12 +290,48 @@ define([
...
+			//create map of names for sorting
+			var descNameMap = {};
+			array.forEach(descriptions, function (desc) {
+				var _name = this._model.authored.getName(desc.value);
+				descNameMap[desc.value] = _name;
+			}, this);
+			descriptions.sort(function (obj1, obj2) {
+				//return obj1.label > obj2.label;
+				if(descNameMap[obj1.value] && descNameMap[obj2.value])
+					return descNameMap[obj1.value].toLowerCase().localeCompare(descNameMap[obj2.value].toLowerCase());
```
Comment (sachingrover211) [inline www/js/con-student.js]: this code should be in model.js as well. The controller should have code to render the field data. The field data manipulation should be part of the model. Like we getAllUnits, the idea is something similar.

**Comment by dharanisakary** [comment]
Pushed changes for Quantity Header

File: www/js/con-student.js (hunk 8 above)
Comment (jonwetzel) [reply www/js/con-student.js]: @dharanisakary I see all changes Sachin requested have been made except this one.  Was there a reason this one is unchanged?

Comment (dharanisakary) [reply www/js/con-student.js]: These changes aren't a part of this pull request. Also, I added new function to model.js to sort the descriptions by variable name. So, after review of those changes, I think we can use that function here somehow. Please let me know your views on this.

Comment (jonwetzel) [reply www/js/con-student.js]: @sachingrover211 Dharani has suggested we wait and make this change after #241.  If that's ok I'll merge this PR and add a note to #241 to do it there.

**Pull request merged**